src/SourceLocator/SourceStubber/ReflectionSourceStubber.php

```diff
@@ -111,11 +115,47 @@ public function generateFunctionStub(string $functionName) : ?StubData
         $this->addDocComment($functionNode, $functionReflection);
         $this->addParameters($functionNode, $functionReflection);
 
+        $extensionName = $functionReflection->getExtensionName() ?: null;
+
         if (! $functionReflection->inNamespace()) {
-            return $this->createStubData($this->generateStub($functionNode), $functionReflection);
+            return $this->createStubData($this->generateStub($functionNode->getNode()), $extensionName);
         }
 
-        return $this->createStubData($this->generateStubInNamespace($functionNode, $functionReflection->getNamespaceName()), $functionReflection);
+        return $this->createStubData($this->generateStubInNamespace($functionNode->getNode(), $functionReflection->getNamespaceName()), $extensionName);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function generateConstantStub(string $constantName) : ?StubData
+    {
+        // Not supported because of resource as value
+        if (in_array($constantName, ['STDIN', 'STDOUT', 'STDERR'], true)) {
+            return null;
+        }
+
+        $constants = get_defined_constants(true);
+
+        $found         = false;
+        $value         = null;
+        $extensionName = null;
+
+        foreach ($constants as $constantExtensionName => $extensionConstants) {
+            if (array_key_exists($constantName, $extensionConstants)) {
+                $found         = true;
+                $value         = $extensionConstants[$constantName];
+                $extensionName = $constantExtensionName !== 'user' ? $constantExtensionName : null;
+                break;
+            }
+        }
+
+        if (! $found) {
```

Comment:
This pattern is quite terrible to maintain/debug: let's please use a `private` method or such instead, if possible.

Reply:
Done.